src/dist.rs

```diff
@@ -0,0 +1,308 @@
+use crate::{default_build_command, default_dist_dir, metadata};
+use anyhow::{ensure, Context, Result};
+use clap::Parser;
+use std::{fs, path::PathBuf, process};
+use wasm_bindgen_cli_support::Bindgen;
+
+/// An helper to generate the distributed package
+///
+/// This structs provide a customizable way to assemble and generate a package
+/// for Wasm.
+///
+/// # Usage
+///
+/// ```rust,no_run
+/// # use std::process;
+/// # use xtask_wasm::{anyhow::Result, clap};
+/// #
+/// # #[derive(clap::Parser)]
+/// # struct Opt {
+/// #     #[clap(subcommand)]
+/// #     cmd: Command,
+/// # }
+/// #
+/// #[derive(clap::Parser)]
+/// enum Command {
+///     Dist(xtask_wasm::Dist),
+/// }
+///
+/// fn main() -> Result<()> {
+///     let opt: Opt = clap::Parser::parse();
+///
+///     match opt.cmd {
+///         Command::Dist(dist) => {
+///             let dist = dist
+///                 .dist_dir_path("dist")
+///                 .static_dir_path("project/static")
+///                 .app_name("project")
+///                 .run_in_workspace(true)
+///                 .run("project")?;
+///
+///             println!("Built at {}", dist.dist_dir.display());
+///         }
+///     }
+///
+///     Ok(())
+/// }
+/// ```
+///
+/// Add a `dist` subcommand to assemble the distributed package for the
+/// `project` crate.
```

Comment:
Not sure why you used backticks here :thinking: 
```suggestion
/// project crate.
```

Reply:
Because i use `project` as the name of the app
I see that can be confusing.